### PR TITLE
Implement Typeform OAuth provider

### PR DESCRIPTION
**Pull request by denis-goncharenko**
This feature allows users to authenticate using their Typeform credentials. Tested in production service and works perfectly.

### CODE REVIEW

File: src/AspNet.Security.OAuth.Typeform/TypeformAuthenticationOptions.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    public TypeformAuthenticationOptions()
+    {
+        ClaimsIssuer = TypeformAuthenticationDefaults.Issuer;
+        CallbackPath = TypeformAuthenticationDefaults.CallbackPath;
+
+        AuthorizationEndpoint = TypeformAuthenticationDefaults.AuthorizationEndpoint;
+        TokenEndpoint = TypeformAuthenticationDefaults.TokenEndpoint;
+        UserInformationEndpoint = TypeformAuthenticationDefaults.UserInformationEndpoint;
+
+        Scope.Add("offline");
```
Comment (martincostello) [inline src/AspNet.Security.OAuth.Typeform/TypeformAuthenticationOptions.cs]: Is this a necessary default?

Comment (denis-goncharenko) [reply src/AspNet.Security.OAuth.Typeform/TypeformAuthenticationOptions.cs]: No, it is just a recommendation if we need a refresh token in response.

**Review by martincostello — APPROVED** [review]

**Comment by kevinchalet** [comment]
Thanks a lot for your contributions, @denis-goncharenko! ❤️ 

Would you be interested in porting these providers to OpenIddict (that we now recommend for new applications)?

[The process is a bit more elaborate](https://documentation.openiddict.com/guides/contributing-a-new-web-provider.html) but it shouldn't require writing custom code as these providers seem fairly standard (you'd just need to update the built-in `MapCustomWebServicesFederationClaims` handler to map the provider-specific claims to their `ClaimTypes` equivalent: https://documentation.openiddict.com/guides/contributing-a-new-web-provider.html#if-the-provider-doesnt-support-standard-openid-connect-userinfo-map-the-provider-specific-claims-to-their-claimtypes-equivalent)

**Review by kevinchalet — APPROVED** [review]

**Comment by denis-goncharenko** [comment]
> Thanks a lot for your contributions, @denis-goncharenko! ❤️
> 
> Would you be interested in porting these providers to OpenIddict (that we now recommend for new applications)?
> 
> [The process is a bit more elaborate](https://documentation.openiddict.com/guides/contributing-a-new-web-provider.html) but it shouldn't require writing custom code as these providers seem fairly standard (you'd just need to update the built-in `MapCustomWebServicesFederationClaims` handler to map the provider-specific claims to their `ClaimTypes` equivalent: https://documentation.openiddict.com/guides/contributing-a-new-web-provider.html#if-the-provider-doesnt-support-standard-openid-connect-userinfo-map-the-provider-specific-claims-to-their-claimtypes-equivalent)

It's quite interesting to me. I am considering porting these providers to OpenIddic in the near future.

**Comment by martincostello** [comment]
Thanks for your contribution @denis-goncharenko - all four of the new providers are now available as part of version 8.1.0.

**Comment by kevinchalet** [comment]
> It's quite interesting to me. I am considering porting these providers to OpenIddict in the near future.

Hey @denis-goncharenko 👋🏻 

Let me know if you're still interested in giving it a try (and please don't hesitate if you need help!).

Cheers.

**Pull request merged**